### PR TITLE
Move from winapi to the windows crate (#137)

### PR DESCRIPTION
**Pull request by Kurtsley**
Here is my attempt at converting to the windows-rs crate. I am not entirely sure how this is supposed to work under the hood so I've made a couple assumptions in the code.

I had to replace the result "check if not 0" code with this block. I'm open to suggestions.

        if result.as_bool() {
            (point.x, point.y)
        } else {
            (0, 0)
        }

This function now needs a VIRTUAL_KEY instead of a u16 so I had to wrap the 0. Again, anyone familiar with the windows-rs crate can correct this. That being said, everything seems to work on my Windows machine.

        keybd_event(
            KEYEVENTF_KEYUP | KEYEVENTF_SCANCODE,
            windows::Win32::UI::Input::KeyboardAndMouse::VIRTUAL_KEY(0),
            scancode,
        );

### CODE REVIEW

**Comment by pentamassiv** [comment]
Wow, last night I wrote an issue for it (https://github.com/enigo-rs/enigo/issues/137) and today you already submitted a PR. Thank you :-)
Since I don't have a Windows machine, I can't test it or say much about it. Merging this would require us to bump the MSRV to 1.64, which was released on Sep 22, 2022.
```
error: package `windows v0.44.0` cannot be built because it requires rustc 1.64 or newer, while the currently active rustc version is 1.58.0
Error: Process completed with exit code 1.
```
I couldn't find out the MSRV of RustDesk (the biggest project using enigo), but in their [build instructions](https://github.com/rustdesk/rustdesk#build-1), they suggest to get Rust via
`curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh`
so their users would have the current version.
My gut feeling would be that it would be acceptable to bump the MSRV to 1.64. What is everyone else's opinion on this?

**Comment by pythoneer** [comment]
> My gut feeling would be that it would be acceptable to bump the MSRV to 1.64. What is everyone else's opinion on this?

I mean if we want to go that route and using `windows-rs` (which i am a big fan of) than we need to support what ever `windows-rs` requires. Does anybody know why the `windows-rs` is so "eager" in having such a new compiler?

File: src/win/win_impl.rs [hunk 1]
```diff
@@ -26,15 +34,22 @@ fn mouse_event(flags: u32, data: u32, dx: i32, dy: i32) {
             })
         },
     };
-    unsafe { SendInput(1, &mut input as LPINPUT, size_of::<INPUT>() as c_int) };
+    unsafe {
+        SendInput(
+            &[input as INPUT],
+            size_of::<INPUT>()
+                .try_into()
+                .expect("Could not convert INPUT to i32"),
```
Comment (pentamassiv) [inline src/win/win_impl.rs:42]: Could you please change this to "Could not convert the size of INPUT to i32"

File: src/win/win_impl.rs [hunk 2]
```diff
@@ -44,16 +59,25 @@ fn keybd_event(flags: u32, vk: u16, scan: u16) {
             input_u
         },
     };
-    unsafe { SendInput(1, &mut input as LPINPUT, size_of::<INPUT>() as c_int) };
+    unsafe {
+        SendInput(
+            &[input as INPUT],
+            size_of::<INPUT>()
+                .try_into()
+                .expect("Could not convert usize to i32"),
```
Comment (pentamassiv) [inline src/win/win_impl.rs:67]: Could you please change this to "Could not convert the size of INPUT to i32"

File: src/win/win_impl.rs [hunk 3]
```diff
@@ -12,10 +20,10 @@ use std::mem::*;
...
 pub struct Enigo;
 
-fn mouse_event(flags: u32, data: u32, dx: i32, dy: i32) {
-    let mut input = INPUT {
-        type_: INPUT_MOUSE,
-        u: unsafe {
+fn mouse_event(flags: MOUSE_EVENT_FLAGS, data: i32, dx: i32, dy: i32) {
+    let input = INPUT {
+        r#type: INPUT_MOUSE,
+        Anonymous: unsafe {
```
Comment (pentamassiv) [inline src/win/win_impl.rs:26]: Can we change this to:
```
let input = INPUT {
        r#type: INPUT_MOUSE,
        Anonymous: INPUT_0 {
            mi: MOUSEINPUT {
                dx,
                dy,
                mouseData: data,
                dwFlags: flags,
                time: 0,
                dwExtraInfo: 0,
            },
        },
    };
```
It compiles and the examples work. I am not familiar writing `unsafe` code though. Are there any reasons why `transmute` needs to be used here?

File: src/win/win_impl.rs [hunk 4]
```diff
@@ -26,15 +34,22 @@ fn mouse_event(flags: u32, data: u32, dx: i32, dy: i32) {
...
-            *input_u.ki_mut() = KEYBDINPUT {
+fn keybd_event(flags: KEYBD_EVENT_FLAGS, vk: VIRTUAL_KEY, scan: u16) {
+    let input = INPUT {
+        r#type: INPUT_KEYBOARD,
+        Anonymous: unsafe {
+            let mut input_u: INPUT_0 = std::mem::zeroed();
+            input_u.ki = KEYBDINPUT {
                 wVk: vk,
                 wScan: scan,
                 dwFlags: flags,
```
Comment (pentamassiv) [inline src/win/win_impl.rs:55]: Similarly, can we change this to:
```
let input = INPUT {
        r#type: INPUT_KEYBOARD,
        Anonymous: INPUT_0 {
            ki: KEYBDINPUT {
                wVk: vk,
                wScan: scan,
                dwFlags: flags,
                time: 0,
                dwExtraInfo: 0,
            },
        },
    };
```
It compiles and the examples work

File: src/win/win_impl.rs (hunk 3 above)
Comment (pentamassiv) [reply src/win/win_impl.rs:26]: Somehow I was unable to comment on the full `let input` statement

**Comment by pentamassiv** [comment]
Thank you for your contribution :)

**Pull request merged**